compiler/infra/ForwardList.hpp

```diff
@@ -0,0 +1,684 @@
+#ifndef FORWARD_LIST_HPP
+#define FORWARD_LIST_HPP
+
+#pragma once
+
+#if 0
```

Comment:
While my first comment addresses the portion of code that is disabled via `#if 0`, there is also the remaining C++03 compatible method definitions from std::forward_list which I skipped implementing, instead commenting out their declarations entirely. Is the preference to remove these?